pages/monsters/[id].vue

```diff
@@ -290,11 +314,36 @@ const abilites = [
   modifier: formatMod(calcMod(monster[ability])),
   save: monster[`${ability}_save`] ?? calcMod(monster[ability]),
 }));
+
+const route = useRoute();
+
+const mode = ref(route.query.mode || 'normal');
+function toggleMode() {
+  switch (mode.value) {
+    case 'compact':
+      mode.value = 'normal';
+      break;
+    default:
+      mode.value = 'compact';
+      break;
+  }
+
+  navigateTo({
+    path: `/monsters/${route.params.id}`,
+    query:
+      mode.value === 'compact'
+        ? {
+            mode: 'compact',
+          }
+        : null,
+  });
+}
```

Comment:
At one point I just had this function navigate to the new route, without the switch block, but because the component was not being re-mounted, the `mode` wasn't being updated from the query param, so I stuck with setting the `mode` ref and then doing the navigate just so there was a copy-able URL for the compact mode